lib/inspec/plugin/v2/filter.rb

```diff
@@ -60,4 +60,14 @@ def validate_plugin_filter_file(_file_version)
       end
     end
   end
+
+  module FilterPredicates
+    def train_plugin_name?(name)
+      name.to_s.start_with?("train-") && ! Inspec::Plugin::V2::PluginFilter.exclude?(name)
+    end
+
+    def inspec_plugin_name?(name)
+      name.to_s.start_with?("inspec-") && ! Inspec::Plugin::V2::PluginFilter.exclude?(name)
```

Comment:
I disagree. The "data" here is that there are two kind of plugins - train plugins and inspec plugins. There are no plans to expand that out into new systems (in fact, we wish we could eliminate the distinction).

When I have three things that repeat, it's definitely worth factoring them out. But this is two and will always be two; if it ever grows beyond 2, well, that's a change for another PR. It's also a very small amount of code duplicated here; any code to curry this out while preserving the original predicates would be larger than the original.

It would just add cleverness to make this factored down - as is, this is dumb and repetitive and obvious, and easy to maintain by people less qualified than either of us.